### ISSUE TITLE
Rule Change: no-empty rule does not ignore empty switch statements with comments

### ISSUE THREAD

**Issue by jaymarvelz** [eslint/eslint#20000]
### What rule do you want to change?

no-empty

### What change do you want to make?

Generate fewer warnings

### How do you think the change should be implemented?

A new default behavior

### Example code

```js
switch (foo) {
  /* empty */
}
```

### What does the rule currently do for this code?

The rule reports an error for the empty switch statement, even though it contains a comment. 

### What will the rule do after it's changed?

After the change, the rule will ignore switch statements that are empty but contain a comment, similar to how it handles other empty blocks.

### Participation

- [x] I am willing to submit a pull request to implement this change.

**Comment by snitin315** [comment]
This is actually the expected behavior. We also a have a test case for this 
https://github.com/eslint/eslint/blob/c36570c6657c2a92dbb4f09a8166a4d9909a091a/tests/lib/rules/no-empty.js#L166-L176

It was added in https://github.com/eslint/eslint/pull/16470, related discussion in https://github.com/eslint/eslint/pull/16096#discussion_r962367178

**Comment by snitin315** [comment]
Confirmed, it is expected behavior. However, I think we can add this in the docs :

https://github.com/eslint/eslint/blob/2309327554acbf011f0d17e7b36fdd68e43adf3a/lib/rules/no-empty.js#L103-L108

**Comment by crimsonjay0** [comment]
What @mdjermanovic mentioned in [#16096 (comment)](https://github.com/eslint/eslint/pull/16096#discussion_r962367178) makes sense — you can’t really provide an autofix for a SwitchStatement unless this issue is addressed. I also don’t see any discussion on why SwitchStatement nodes are excluded from this behavior.

**Comment by kirkwaiblinger** [comment]
It seems that this behavior (empty switches are flagged regardless of presence of comment) has been present since the exemption for comments was initially added (commit https://github.com/eslint/eslint/commit/550bbe1929364dfb77f821c34107d530722e83ca). There is no mention of an intentional omission thereof for switches in the PR (https://github.com/eslint/eslint/pull/2043), and no such mention in associated discussion in (https://github.com/eslint/eslint/issues/1841). 

The testing change in https://github.com/eslint/eslint/pull/16470/files#diff-4945c2daa6580f5e320065eb7489586a17a820f3c58b8e0439499f147aaa98fdR134-R142 appears to simply assert the existing behavior, not provide justification for it.

Based on this information, this _appears_ to be a simple oversight due to `SwitchStatement` being checked in a separate selector than `BlockStatement`, rather than an intentional decision.

**Comment by mdjermanovic** [comment]
> It seems that this behavior (empty switches are flagged regardless of presence of comment) has been present since the exemption for comments was initially added (commit [550bbe1](https://github.com/eslint/eslint/commit/550bbe1929364dfb77f821c34107d530722e83ca)). There is no mention of an intentional omission thereof for switches in the PR ([#2043](https://github.com/eslint/eslint/pull/2043)), and no such mention in associated discussion in ([#1841](https://github.com/eslint/eslint/issues/1841)).

Indeed, there's no mention of `switch` statements in the PR that introduced allowing empty blocks with comments, nor in the original issue. I think this may have been an oversight, though if we are not sure if this was intentional or not, it might be safer to add this behavior behind an option. @eslint/eslint-team thoughts?

> The testing change in https://github.com/eslint/eslint/pull/16470/files#diff-4945c2daa6580f5e320065eb7489586a17a820f3c58b8e0439499f147aaa98fdR134-R142 appears to simply assert the existing behavior, not provide justification for it.

Correct, the commit was about adding suggestions to this rule, and the test was just asserting the existing behavior.

**Comment by nzakas** [comment]
I think we can treat this as a bug and just fix it without adding a new option. It seems like an oversight to me.

**Comment by mdjermanovic** [comment]
Agreed. @jaymarvelz a PR to implement this change is welcome:



> ### What will the rule do after it's changed?
> After the change, the rule will ignore switch statements that are empty but contain a comment, similar to how it handles other empty blocks.
